network/src/counters.rs

```diff
@@ -557,3 +561,20 @@ pub fn network_application_outbound_traffic(
         ])
         .observe(size as f64);
 }
+
+/// Time it takes to perform message serialization and deserialization
+pub static NETWORK_APPLICATION_SERIALIZATION_METRIC: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
```

Comment:
The default buckets might not be a good choice here.

Reply:
Yeah -- I did wonder about that. 😄 From some forge runs, they looked okay. I can send out a separate PR if we feel we need more accuracy. I'm less worried about precision and more about identifying bad behaviours (i.e., if things start to blow up).